### PR TITLE
use pair-wise testing on PR validation

### PR DESCRIPTION
**Pull request by jmarolf**
Same approach as was done in this PR https://github.com/dotnet/roslyn/pull/57586 applied to integration tests.

| Configuration | Bitness | Runtime   | Editor | Run on PR |
|:--------------|---------|-----------|--------|-----------|
| Debug         | x86     | Framework | Legacy | No        |
| Debug         | amd64   | Framework | Legacy | Yes       |
| Release       | x86     | Framework | Legacy | Yes       |
| Release       | amd64   | Framework | Legacy | No        |
| Debug         | amd64   | Framework | LSP    | Yes       |
| Debug         | amd64   | Core      | Legacy | Yes       |
| Release       | amd64   | Core      | Legacy | No        |

### CODE REVIEW

**Comment by jmarolf** [comment]
merged ci run [here](https://dev.azure.com/dnceng/public/_build/results?buildId=1462799&view=results)

File: azure-pipelines-integration.yml [hunk 1]
```diff
@@ -14,35 +14,65 @@ pr:
 - features/*
 - demos/*
 
-variables:
-  - name: XUNIT_LOGS
-    value: $(Build.SourcesDirectory)\artifacts\log\$(_configuration)
-
 jobs:
-- job: VS_Integration
+- ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
```
Comment (Cosifne) [inline azure-pipelines-integration.yml:18]: Could you remind me when will 'Build.Reason' not equals to 'PullRequest'?

**Review by Cosifne — APPROVED** [review]
LGTM

File: azure-pipelines-integration-corehost.yml [hunk 2]
```diff
@@ -18,27 +18,34 @@ pr:
...
         oop64bit: true
         oopCoreClr: true
+
+- ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
+  - job: VS_Integration_CoreHost_Release
+    pool:
+      name: NetCore1ESPool-Public
+      demands: ImageOverride -equals $(queueName)
+    timeoutInMinutes: 135
+    variables:
```
Comment (RikkiGibson) [inline azure-pipelines-integration-corehost.yml:43]: What's the effect of pushing these variables from the root into the individual jobs?

**Review by RikkiGibson — APPROVED** [review]

**Comment by RikkiGibson** [comment]
Related to #57313

**Pull request merged**